storage/src/backend/localfs.rs

```diff
@@ -435,6 +437,37 @@ impl Prefetcher {
         })
     }
 
+    #[cfg(target_os = "macos")]
+    fn do_readahead(&self) -> Result<()> {
+        info!("starting localfs blob readahead");
+
+        for &(offset, len, zero) in self.records.iter() {
+            let end: u64 = offset
+                .checked_add(len as u64)
+                .ok_or_else(|| einval!("invalid length"))?;
+            if offset > self.blob_size || end > self.blob_size || zero != 0 {
+                return Err(einval!(format!(
+                    "invalid readahead entry ({}, {}), blob size {}",
+                    offset, len, self.blob_size
+                )));
+            }
+
+            unsafe {
```

Comment:
How about using `posix_fadvise` that is supposed to work on both Linux and Unix (macos).
https://man7.org/linux/man-pages/man2/posix_fadvise.2.html

Reply:
The `posix_fadvise`  not found in macos. It's sad..